sw/device/lib/dif/dif_aon_timer.h

```diff
@@ -0,0 +1,272 @@
+    uint32_t bite_threshold, bool pause_in_sleep, bool lock);
+
+/** Stops Always-On Timer (watchdog timer).
+ *
+ * Stops the timer. Configuration is not touched, and can be restarted via
+ * `dif_aon_timer_watchdog_restart`.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_stop(
+    const dif_aon_timer_t *handle);
+
+/** Restarts Always-On Timer (watchdog timer).
+ *
+ * Clears the counter and restarts the timer using the existing configuration.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_restart(
+    const dif_aon_timer_t *handle);
+
+/** Retrieves Always-On Timer (watchdog timer) tick count.
+ *
+ * @param handle An Always-On Timer handle.
+ * @param[out] count Current timer tick count.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_watchdog_get_count(
+    const dif_aon_timer_t *handle, uint32_t *count);
+
+/** Clears Always-On Timer (watchdog timer).
+ *
+ * This function must be called periodically to satisfy "Bite" and "Bark"
+ * thresholds.
+ *
+ * @param handle A Always-On Timer handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_pet_watchdog(
+    const dif_aon_timer_t *handle);
+
+/**
+ * Checks whether this Always-On Timer (watchdog timer) is locked.
+ *
+ * @param handle A Always-On Timer handle.
+ * @param[out] is_locked Out-param for the locked state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_is_locked(
+    const dif_aon_timer_t *handle, bool *is_locked);
+
+/**
+ * Returns whether a particular interrupt is currently pending.
+ *
+ * @param handle A Always-On Timer handle.
+ * @param irq An interrupt type.
+ * @param[out] is_pending Out-param for whether the interrupt is pending.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_irq_is_pending(
+    const dif_aon_timer_t *handle, dif_aon_timer_irq_t irq, bool *is_pending);
+
+/**
+ * Forces a particular interrupt, causing it to be serviced as if hardware had
+ * asserted it.
+ *
+ * @param handle A Always-On Timer handle.
+ * @param irq An interrupt type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aon_timer_result_t dif_aon_timer_irq_force(const dif_aon_timer_t *handle,
+                                               dif_aon_timer_irq_t irq);
+
+/**
+ * Retrieves the maximal Always-On Timer threshold in ticks.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @return Maximal timer threshold in ticks.
+ */
+DIF_WARN_UNUSED_RESULT
+uint32_t dif_aon_timer_get_max_threshold(void);
```

Comment:
Not sure if this function is useful, or we should just assume 32-bit.

Reply:
Given that the interface elsewhere here has assumed a 32bit counter I think that this isn't necessary.

Reply:
Done.